### ISSUE TITLE
std::numeric_limits<T>::max() causing compilation errors on windows

### ISSUE THREAD

**Issue by Kanagawa-Wave** [KhronosGroup/Vulkan-Hpp#1477]
The sample code (for example in `utils.hpp` and `utils.cpp`)contains a lot of calls to `std::numeric_limits<T>::max()` which seems to not compile in windows platform (at least on my PC it is not compiling), compiler logs below:
```
utils.hpp(354,7): Warning C4003 : not enough arguments for function-like macro invocation 'max'
utils.cpp(433,7): Warning C4003 : not enough arguments for function-like macro invocation 'max'
utils.cpp(433,7): Error C2589 : '(': illegal token on right side of '::'
utils.cpp(433,7): Error C2062 : type 'unknown-type' unexpected
utils.cpp(433,7): Error C2059 : syntax error: ')'
```
It seems like MSVC have taken the `max()` here as this macro `#define max(a,b) (((a) > (b)) ? (a) : (b))` defined in `windef.h`, yet I am not sure where is this header included, my guess would be glfw maybe?

Long story short, a simple fix will fix this problem:
`std::numeric_limits<T>::max()` -> `(std::numeric_limits<T>::max)()`

Env:
Windows SDK 10.0
MSVC C++17

**Comment by asuessenbach** [comment]
I think, the "standard procedure" to prevent that error is to globally #define NOMINMAX.

**Comment by asuessenbach** [comment]
Seems to be resolved. Please re-open if not.
